easybuild/easyblocks/o/openmpi.py

```diff
@@ -0,0 +1,120 @@
+            if enable_opt:
+                regex = re.compile('--(disable|enable)-%s' % key)
+            else:
+                regex = re.compile('--(with|without)-%s' % key)
+
+            return not bool(regex.search(self.cfg['configopts']))
+
+        config_opt_names = [
+            # suppress failure modes in relation to mpirun path
+            'mpirun-prefix-by-default',
+            # build shared libraries
+            'shared',
+        ]
+
+        for key in config_opt_names:
+            if config_opt_unused(key, enable_opt=True):
+                self.cfg.update('configopts', '--enable-%s' % key)
+
+        # check whether VERBS support should be enabled
+        if config_opt_unused('verbs'):
+
+            # auto-detect based on available OS packages
+            verbs = False
+            for osdep in ['libibverbs-dev', 'libibverbs-devel', 'rdma-core-devel']:
+                if check_os_dependency(osdep):
+                    verbs = True
+                    break
+
+            if verbs:
+                self.cfg.update('configopts', '--with-verbs')
+            else:
+                self.cfg.update('configopts', '--without-verbs')
+
+        # handle dependencies
+        for dep in ['CUDA', 'hwloc', 'libevent', 'PMIx', 'UCX']:
+            if config_opt_unused(dep.lower()):
+                dep_root = get_software_root(dep)
+                if dep_root:
+                    self.cfg.update('configopts', '--with-%s=%s' % (dep.lower(), dep_root))
+
+        super(EB_OpenMPI, self).configure_step()
+
+    def sanity_check_step(self):
+        """Custom sanity check for OpenMPI."""
+
+        bin_names = ['mpicc', 'mpicxx', 'mpif90', 'mpifort', 'mpirun', 'ompi_info', 'opal_wrapper', 'orterun']
+        bin_files = [os.path.join('bin', x) for x in bin_names]
+
+        shlib_ext = get_shared_lib_ext()
+        lib_names = ['mpi_mpifh', 'mpi', 'ompitrace', 'open-pal', 'open-rte']
+        lib_files = [os.path.join('lib', 'lib%s.%s' % (x, shlib_ext)) for x in lib_names]
+
+        inc_names = ['mpi-ext', 'mpif-config', 'mpif', 'mpi', 'mpi_portable_platform']
+        inc_files = [os.path.join('include', x + '.h') for x in inc_names]
+
+        custom_paths = {
+            'files': bin_files + inc_files + lib_files,
+            'dirs': [],
+        }
+
+        # make sure MPI compiler wrappers pick up correct compilers
+        expected = {
+            'mpicc': os.getenv('CC', 'gcc'),
+            'mpicxx': os.getenv('CXX', 'g++'),
+            'mpifort': os.getenv('FC', 'gfortran'),
+            'mpif90': os.getenv('F90', 'gfortran'),
+        }
+        # actual pattern for gfortran is "GNU Fortran"
+        for key in ['mpifort', 'mpif90']:
+            if expected[key] == 'gfortran':
+                expected[key] = "GNU Fortran"
+
+        custom_commands = ["%s --version | grep '%s'" % (key, expected[key]) for key in sorted(expected.keys())]
```

Comment:
Is this verified to work with Intel and PGI too?

Reply:
It's exactly for those compilers this was added, see https://github.com/easybuilders/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.10.2-PGI-16.4-GCC-5.3.0-2.26.eb#L33 and https://github.com/easybuilders/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.10.3-iccifort-2016.3.210-GCC-5.4.0-2.26.eb#L32